.github/workflows/python-app.yml

```diff
@@ -0,0 +1,40 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python application
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest Flake8-pyproject
```

Comment:
I would recommend to use a fixed version of flake8. Using the latest version has a probability of adding more checkers, that then fail on existing code, which you do not want unless you do the update manually and fix the problems / deactivate the checkers

Reply:
The easiest solution, that also keeps the supported version as part of the code would be a dev-requirements.txt file, that can then be parsed by pip to install the exact dependencies: `pip install --requirement dev-requirements.txt`

Reply:
Done